plugins/modules/fabric_virtual_networks_workflow_manager.py

```diff
@@ -0,0 +1,2950 @@
+            return True
+
+        if len(fabric_site_ids) != len(fabric_ids_in_ccc):
+            return True
+
+        for fabric_id in fabric_site_ids:
+            if fabric_id not in fabric_ids_in_ccc:
+                return True
+
+        anchor_site = vn_details.get("anchored_site_name")
+        if anchor_site:
+            site_id = self.get_site_id(anchor_site)
+
+            if not site_id:
+                msg = "Given Anchor site '{0}' not  present in Cisco Catalyst Center.".format(anchor_site)
+                self.log(msg, "ERROR")
+                return False
+            try:
+                anchor_fabric_id = self.get_fabric_site_id(anchor_site, site_id)
+            except Exception as e:
+                anchor_fabric_id = self.get_fabric_zone_id(anchor_site, site_id)
+
+            if anchor_fabric_id and anchor_fabric_id != vn_in_ccc.get("anchoredSiteId"):
+                return True
+
+        return False
+
+    def update_payload_vn(self, vn_details, vn_in_ccc):
+        """
+        Constructs an update payload for a virtual network based on the provided details and its current
+        configuration in the Cisco Catalyst Center.
+        Args:
+            self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+            vn_details (dict): A dictionary containing details about VN to be updated, including:
+                - vn_name (str): The name of the virtual network.
+                - fabric_site_locations (list): A list of fabric site locations associated with virtual network.
+                - anchored_site_name (str): The name of the anchored site.
+            vn_in_ccc (dict): A dictionary representing the current state of the virtual network in the
+                            Cisco Catalyst Center, which includes:
+                - id (str): The identifier of the existing virtual network.
+                - anchoredSiteId (str): The ID of anchored site currently associated with the virtual network.
+        Returns:
+            dict: A dictionary representing the payload for updating the virtual network.
+        Description:
+            This function constructs a payload for updating a virtual network by gathering relevant details
+            from the provided `vn_details` and the current configuration in `vn_in_ccc`.
+            The function returns the constructed update payload for the virtual network, which includes all
+            necessary identifiers and configurations needed for the update operation.
+        """
+
+        vn_name = vn_details.get("vn_name")
+        update_vn_payload = {
+            "id": vn_in_ccc.get("id"),
+            "virtualNetworkName": vn_name
+        }
+        fabric_locations = vn_details.get("fabric_site_locations")
+        fabric_site_ids = []
+
+        if isinstance(fabric_locations, list) and not fabric_locations:
```

Comment:
This check is not clear.. 
fabric_locations is a list but not there?
Do you meant to say
```
if not fabric_locations or not isinstance(fabric_locations, list)
     update_vn_payload["fabricIds"] = []
else:
      fabric_site_ids = self.get_fabric_ids(fabric_locations)
      update_vn_payload["fabricIds"] = fabric_site_ids
```